### PR TITLE
ASL-4247 - Added github auth to the npm build

### PR DESCRIPTION
**Pull request by tomrowsell86**
Also parameterised branch name in the node semver increment shared workflow

### CODE REVIEW

File: .github/workflows/semver-check-increment-npm.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 name: 'SemVer label Check and Increment package.json - npm'
 on:
-  workflow_call:
+  workflow_call: 
+    inputs:
+      mainLineBranchName:
```
Comment (LiamMacP) [inline .github/workflows/semver-check-increment-npm.yml:5]: We seem to have the standard of lowercase split by underscores, could this be changed to fit this?

File: .github/actions/npm-build-project/action.yml [hunk 2]
```diff
@@ -19,10 +22,14 @@ runs:
       with:
         node-version: ${{ inputs.node_version }}
         cache: 'npm'
-
+        registry-url: 'https://npm.pkg.github.com'
```
Comment (LiamMacP) [inline .github/actions/npm-build-project/action.yml:25]: Would this allow normal packages from the NPM registry to be pulled to?

File: .github/actions/npm-build-project/action.yml [hunk 3]
```diff
@@ -19,10 +22,14 @@ runs:
...
+        scope: '@ukhomeoffice'
+        
     - name: Install npm dependencies
       run: npm --loglevel warn ${{ inputs.install_command || 'ci' }}
```
Comment (LiamMacP) [inline .github/actions/npm-build-project/action.yml:29]: Following this: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#use-private-packages

Do we need to add the following flag? `--ignore-scripts`

File: .github/workflows/test-npm.yml [hunk 4]
```diff
@@ -54,11 +55,12 @@ jobs:
           submodules: true
 
       - name: Build project
-        uses: UKHomeOffice/sas-github-workflows/.github/actions/npm-build-project@v1
+        uses: UKHomeOffice/sas-github-workflows/.github/actions/npm-build-project@TR_BRANCH1
```
[the remark this answers is not in the document]
Comment (LiamMacP) [reply .github/workflows/test-npm.yml:57]: Could this be a separate PR if required? This is marked as a minor change so I believe v1 would still be applicable.

**Review by LiamMacP — CHANGES_REQUESTED** [review]
Just a couple of little bits that need looking at 🙂

File: .github/workflows/test-npm.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomrowsell86) [reply .github/workflows/test-npm.yml:57]: Yeah I will do that as a separate PR then I'll clean up the tags as part of that PR.

File: .github/workflows/semver-check-increment-npm.yml (hunk 1 above)
Comment (tomrowsell86) [reply .github/workflows/semver-check-increment-npm.yml:5]: Had a look elsewhere in the repo and I think that the inputs use CAML case. Although given the yaml properties are generally snake case I can see why that looks a bit wrong!

Comment (nickmcmahon01) [reply .github/workflows/semver-check-increment-npm.yml:5]: there are two standards in the repo it seems, one for workflows and one for actions
https://github.com/UKHomeOffice/sas-github-workflows/blob/main/.github/workflows/actions-check-dist.yml#L12
vs
https://github.com/UKHomeOffice/sas-github-workflows/blob/main/.github/actions/docker-publish/action.yml#L19

**Pull request merged**